packages/atoms/src/Radio.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState } from 'react';
+import {
+    NativeStyleObject,
+    NativeStyleUtils,
+    prepareNativeStyle,
+    useNativeStyles,
+} from '@trezor/styles';
+import { Pressable, View } from 'react-native';
+
+type RadioValue = string | number | undefined;
+
+interface RadioProps {
+    value: string;
+    isChecked?: boolean;
+    onPress?: (value: RadioValue) => void;
+    isDisabled?: boolean;
+    style?: NativeStyleObject;
+}
+
+type RadioStyleProps = {
+    isChecked?: boolean;
+    isDisabled?: boolean;
+};
+
+const getRadioBorderColor = (
+    utils: NativeStyleUtils,
+    { isChecked, isDisabled }: RadioStyleProps,
+) => {
+    if (isChecked && !isDisabled) {
+        return utils.colors.green;
+    }
+    return utils.colors.gray400;
+};
+
+const radioStyle = prepareNativeStyle<RadioStyleProps>((utils, { isChecked, isDisabled }) => ({
+    height: 24,
+    width: 24,
+    backgroundColor: isDisabled ? utils.colors.gray400 : utils.colors.white,
+    flexDirection: 'row',
+    alignItems: 'center',
+    justifyContent: 'center',
+    borderRadius: utils.borders.radii.round,
+    borderWidth: isChecked ? utils.borders.widths.lg : utils.borders.widths.md,
+    borderColor: getRadioBorderColor(utils, { isChecked, isDisabled }),
```

Comment:
please use extend feature of styles:

```tsx
borderColor: utils.colors.gray400,
extend: {
	condition: isChecked && !isDisabled,
	style: { borderColor: utils.colors.green }
}
```

Reply:
And if there are several conditions, there will be more extend blocks? Something like `getRadioBorderColor()` in that case would be more clear here + easier to reuse.

Reply:
For several conditions I would in fact use that helper function ... At least from my point of view it is more readable to have such helper function encapsulating some context than 2 or more extends..